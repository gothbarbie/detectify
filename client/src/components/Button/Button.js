import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const DefaultButton = styled.button`
  font-size: 1.8rem;
  color: #fff;
  font-weight: 400;
  background-color: transparent;
  border-radius: 2px;
  border: none;
  height: 4rem;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`

const RedButton = styled(DefaultButton)`
  background-color: #ba3333;
`

const BlueButton = styled(DefaultButton)`
  background-color: #7f98b7;
`

const Button = ({ children, type, onClick }) => {
  return type === 'red' ? (
    <RedButton onClick={onClick}>{children}</RedButton>
  ) : (
    <BlueButton onClick={onClick}>{children}</BlueButton>
  )
}

Button.defaultProps = {
  type: 'blue',
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.string,
  onClick: PropTypes.func.isRequired,
}

export default Button
