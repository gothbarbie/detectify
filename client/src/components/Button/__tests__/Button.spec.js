import React from 'react'
import { shallow } from 'enzyme'

import Button from '../Button'

describe('Components/Button', () => {
  it('renders default (blue)', () => {
    const props = {
      onClick: jest.fn(),
    }
    const wrapper = shallow(<Button {...props}>Text</Button>)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders red', () => {
    const props = {
      onClick: jest.fn(),
      type: 'red',
    }
    const wrapper = shallow(<Button {...props}>Text</Button>)

    expect(wrapper).toMatchSnapshot()
  })
})
