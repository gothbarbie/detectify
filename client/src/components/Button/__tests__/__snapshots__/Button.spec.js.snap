// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button renders default (blue) 1`] = `
<Styled(styled.button)
  onClick={[MockFunction]}
>
  Text
</Styled(styled.button)>
`;

exports[`Components/Button renders red 1`] = `
<Styled(styled.button)
  onClick={[MockFunction]}
>
  Text
</Styled(styled.button)>
`;
